src/data/Attempt.ts

```diff
@@ -34,6 +34,10 @@ export interface Attempt {
   solution?: AlgString; // TODO
   penalties?: Penalty[]; // TODO
 
+  // simplified way of indicating Penalty, users must write their own penalty logic
+  // Ok = 0, +2 = 1, DNF = -1 (csTimer style)
+  penaltyCode?: number;
```

Comment:
I think this is best left for another discussion. This design would take more discussion because:

- It introduces two contradictory ways of doing things (which every client would have to write logic to support).
- It uses magic numbers that are not 100% obvious without context.

Reply:
`It introduces two contradictory ways of doing things (which every client would have to write logic to support).`
Yes, but I thought it would be better for simple use cases
`It uses magic numbers that are not 100% obvious without context.`
We can just mention this in the documentation